node/src/tasks/msp_move_bucket.rs

```diff
@@ -88,230 +91,560 @@ where
+        }
+
+        let call = storage_hub_runtime::RuntimeCall::FileSystem(
+            pallet_file_system::Call::msp_respond_move_bucket_request {
+                bucket_id,
+                response: BucketMoveRequestResponse::Rejected,
+            },
+        );
+
+        self.storage_hub_handler
+            .blockchain
+            .submit_extrinsic_with_retry(
+                call,
+                RetryStrategy::default()
+                    .with_max_retries(3)
+                    .with_max_tip(10.0)
+                    .with_timeout(Duration::from_secs(
+                        self.storage_hub_handler
+                            .provider_config
+                            .extrinsic_retry_timeout,
+                    )),
+                false,
+            )
+            .await
+            .map_err(|e| {
+                anyhow!(
+                    "Failed to submit move bucket rejection after 3 retries: {:?}",
+                    e
+                )
+            })?;
+
+        Ok(())
+    }
+
+    async fn accept_bucket_move(&self, bucket_id: BucketId) -> anyhow::Result<()> {
+        info!(
+            target: LOG_TARGET,
+            "MSP: accepting move bucket request for bucket {:?}",
+            bucket_id.as_ref(),
+        );
+
+        let call = storage_hub_runtime::RuntimeCall::FileSystem(
+            pallet_file_system::Call::msp_respond_move_bucket_request {
+                bucket_id,
+                response: BucketMoveRequestResponse::Accepted,
+            },
+        );
+
+        info!(
+            target: LOG_TARGET,
+            "MSP: accepting move bucket request for bucket {:?}",
+            bucket_id.as_ref(),
+        );
+
+        self.storage_hub_handler
+            .blockchain
+            .submit_extrinsic_with_retry(
+                call,
+                RetryStrategy::default()
+                    .with_max_retries(3)
+                    .with_max_tip(10.0)
+                    .with_timeout(Duration::from_secs(
+                        self.storage_hub_handler
+                            .provider_config
+                            .extrinsic_retry_timeout,
+                    )),
+                false,
+            )
+            .await
+            .map_err(|e| {
+                anyhow!(
+                    "Failed to submit move bucket acceptance after 3 retries: {:?}",
+                    e
+                )
+            })?;
+
+        Ok(())
+    }
+
+    async fn download_file(
+        &self,
+        file: &shc_indexer_db::models::File,
+        bucket: &BucketId,
+    ) -> anyhow::Result<()> {
+        let file_metadata = file.to_file_metadata(bucket.as_ref().to_vec());
+        let file_key = file_metadata.file_key::<HashT<StorageProofsMerkleTrieLayout>>();
+
+        info!(
+            target: LOG_TARGET,
+            "MSP: downloading file {:?}",
+            file_key,
+        );
+
+        let chunks_count = file_metadata.chunks_count();
```

Comment:
I would just move this before the for loop since we only use this variable at that point.